### ISSUE TITLE
Liability initial balance issues

### ISSUE THREAD

**Issue by ms32035** [firefly-iii/firefly-iii#1939]
**Bug description**
I am running Firefly III version 4.7.9

When a liability is created, an account for initial balance with type `Initial balance account` is created
1) when calling /api/v1/account account type for this account is returned as empty string, causing validation errors
2) when the liability is deleted, the value of `accounts.deleted_at` is not updated

**Steps to reproduce**
1. Create liability
2. call GET on /api/v1/accounts
3. Delete the liability
4. check sql table

**Expected behavior**
properly returned account type by the api. There's more discrepancies:
YAML
```
        type:
          type: string
          format: string
          example: "asset"
          enum:
            - asset
            - expense
            - import
            - revenue
            - cash
            - liabilities
```
DB Table
```
+-------------------------+
| type                    |
+-------------------------+
| Default account         |
| Cash account            |
| Asset account           |
| Expense account         |
| Revenue account         |
| Initial balance account |
| Beneficiary account     |
| Import account          |
| Loan                    |
| Reconciliation account  |
| Debt                    |
| Mortgage                |
| Credit card             |
+-------------------------+
```

**Extra info**
Please add extra info here, such as OS, browser, and the output from the /debug page of your Firefly III installation (click the version at the bottom).

**Bonus points**
Earn bonus points by:

- Post a stacktrace from your log files
- Add a screenshot

**Comment by JC5** [comment]
Thanks for letting me know!

I've pushed fixes in the API docs and the API, so that the initial balance accounts will now report `initial-balance` when requested through the API. The API docs will update in a few hours, the demo site has been updated.

As for the database table, some of these are no longer in use so they are not mentioned in the documentation: "Default account" and "Beneficiary account" are long deprecated and "Import account" was never used.
